src/library/searchquery.cpp

```diff
@@ -107,6 +107,10 @@ NotNode::NotNode(QueryNode* pNode)
         : m_pNode(pNode) {
 }
 
+NotNode::~NotNode() {
+    delete m_pNode;
```

Comment:
m_pnode is passed in through the constructor, does this object own that pointer?  If not, it shouldn't delete it.